src-tauri/src/state/reducers/credential_offer.rs

```diff
@@ -52,8 +55,37 @@ pub async fn read_credential_offer(state: &mut AppState, action: Action) -> Resu
         None
     };
 
+    let credential_issuer_metadata = wallet
+        .get_credential_issuer_metadata(credential_issuer_url.clone())
+        .await
+        .ok();
```

Comment:
Because of this code `test_qr_code_scanned_read_credential_offer` is failing. This is because that test relied on the fact that previously, in case the credential offer by value, there would not be a GET request to the issuer's metadata endpoint. With this addition the test will timeout because it cannot fetch any metadata (because it doesn't exists for that test). 
The code above your code:
```
    // Get the credential issuer metadata.
    let credential_issuer_metadata = if credential_offer
        .credentials
        .iter()
        .any(|credential| matches!(credential, CredentialsObject::ByReference(_)))
    {
        wallet
            .get_credential_issuer_metadata(credential_issuer_url.clone())
            .await
            .ok()
    } else {
        None
    };
```
ensured that there would be no HTTP request made by the `wallet` if not needed. 

However, since with your current update, we **do** need that metadata, regardless of whether the credential offer is by value or by reference. Therefore you can completely remove my code in lines 45 until 56. You can also completely remove `test_qr_code_scanned_read_credential_offer` since it is not 'supposed' to work anymore in it's current form.

So TL;DR:
* remove lines 45 until 56 in this file
* remove the `test_qr_code_scanned_read_credential_offer` test